cpp/src/arrow/ipc/writer.cc

```diff
@@ -1070,6 +1070,9 @@ class ARROW_EXPORT IpcFormatWriter : public RecordBatchWriter {
   Status WriteRecordBatch(
       const RecordBatch& batch,
       const std::shared_ptr<const KeyValueMetadata>& custom_metadata) override {
+    if (closed_) {
+      return Status::Invalid("Destination already closed");
+    }
```

Comment:
Do we need similar check in `WriteTable` too?

Reply:
Good question. 
It appears that WriteTable ultimately calls WriteRecordBatch. This appears to be documented as well, in that the documentation for RecordBatchWriter::WriteTable is that a a table is written by creating a sequence of record batches. So testing WriteTable doesn't seem to be necessary.

Reply:
Thanks. I see.